Source/SteveSloth/Public/MyEnemyAIIdleState.h

```diff
@@ -0,0 +1,41 @@
+/****************************************************************************************
+* Copyright: SteveSloth
+ * Name: Aidan MacLeod
+ * Script: MyEnemyAIIdleState.h
+ * Date: April 29. 2024
+ * Description: Idle State Class for all enemies to inherit from
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+#pragma once
+
+#include "CoreMinimal.h"
+#include "GameFramework/Character.h"
+#include "MyEnemyAIIdleState.generated.h"
+
+UCLASS()
+class STEVESLOTH_API AMyEnemyAIIdleState : public ACharacter
+{
+	GENERATED_BODY()
+private:
+	bool IsIdle;
+	bool HasDetectedPlayer;
```

Comment:
rename to IsPlayerDetected
and maybe write a getter for this variable depending on how you're planning on swapping states